rust/alloc/vec/mod.rs

```diff
@@ -2160,6 +2383,38 @@ impl<T: Clone, A: Allocator> Vec<T, A> {
         self.spec_extend(other.iter())
     }
 
+    /// Tries to clone and append all elements in a slice to the `Vec`.
+    ///
+    /// Iterates over the slice `other`, clones each element, and then appends
+    /// it to this `Vec`. The `other` vector is traversed in-order.
+    ///
+    /// Note that this function is same as [`extend`] except that it is
+    /// specialized to work with slices instead. If and when Rust gets
+    /// specialization this function will likely be deprecated (but still
+    /// available).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut vec = vec![1];
+    /// vec.try_extend_from_slice(&[2, 3, 4]).unwrap();
+    /// assert_eq!(vec, [1, 2, 3, 4]);
+    /// ```
+    ///
+    /// [`extend`]: Vec::extend
+    #[stable(feature = "kernel", since = "1.0.0")]
+    pub fn try_extend_from_slice(&mut self, other: &[T]) -> Result<(), TryReserveError> {
+        // TODO: another method to have the option for `try_reserve_exact`?
+        self.try_reserve(other.len())?;
+
+        // TODO: be smarter. Also avoids bringing the `spec_extend` module.
```

Comment:
Specialization is *the* approach to be "smart" currently. I think we should just ungate `spec_extend` module and gate `SpecExtend::spec_extend` instead, and add a `try_spec_extend` method to `SpecExtend`.

Reply:
I didn't mean it in a bad way, i.e. I was just trying to say that I am doing pushes one by one since it is the simplest implementation to get things rolling. I have a similar comment elsewhere.

Reply:
Hmm... To put it through `spec_extend` I had to add quite a few things, but let's see.